pkg/ingester/ingester_ingest_storage_test.go

```diff
@@ -41,6 +42,242 @@ import (
 	"github.com/grafana/mimir/pkg/util/validation"
 )
 
+func TestIngester_Start(t *testing.T) {
+	util_test.VerifyNoLeak(t)
+
+	t.Run("should replay the partition at startup and then join the ingesters and partitions ring", func(t *testing.T) {
+		var (
+			ctx                = context.Background()
+			cfg                = defaultIngesterTestConfig(t)
+			reg                = prometheus.NewRegistry()
+			fetchRequestsCount = atomic.NewInt64(0)
+			series1            = mimirpb.PreallocTimeseries{TimeSeries: &mimirpb.TimeSeries{
+				Labels:  mimirpb.FromLabelsToLabelAdapters(labels.FromStrings(labels.MetricName, "series_1")),
+				Samples: []mimirpb.Sample{{TimestampMs: 1000, Value: 10}},
+			}}
+			series2 = mimirpb.PreallocTimeseries{TimeSeries: &mimirpb.TimeSeries{
+				Labels:  mimirpb.FromLabelsToLabelAdapters(labels.FromStrings(labels.MetricName, "series_2")),
+				Samples: []mimirpb.Sample{{TimestampMs: 1000, Value: 10}},
+			}}
+		)
+
+		const expectedSeriesToReplay = 2
+
+		// Configure the ingester to frequently run its internal services.
+		cfg.UpdateIngesterOwnedSeries = true
+		cfg.OwnedSeriesUpdateInterval = 100 * time.Millisecond
+		cfg.BlocksStorageConfig.TSDB.HeadCompactionInterval = 100 * time.Millisecond
+		cfg.ActiveSeriesMetrics.UpdatePeriod = 100 * time.Millisecond
+		cfg.limitMetricsUpdatePeriod = 100 * time.Millisecond
+
+		// Create the ingester.
+		overrides, err := validation.NewOverrides(defaultLimitsTestConfig(), nil)
+		require.NoError(t, err)
+		ingester, kafkaCluster, watcher := createTestIngesterWithIngestStorage(t, &cfg, overrides, reg)
+
+		// Mock the Kafka cluster to:
+		// - Count the Fetch requests.
+		// - Mock the ListOffsets response, returning the offset expected once the ingester can be
+		//   considered having successfully catched up.
```

Comment:
nit: catched -> caught.

Reply:
Fixed in https://github.com/grafana/mimir/pull/7486/commits/2911937a5ea3cbf21c42fa4f6e59c433251ef939